src/devices/machine/ds1207.cpp

```diff
@@ -96,12 +96,12 @@ void ds1207_device::device_start()
 
 void ds1207_device::nvram_default()
 {
-	memset(m_unique_pattern, 0, sizeof(m_unique_pattern));
-	memset(m_identification, 0, sizeof(m_identification));
-	memset(m_security_match, 0, sizeof(m_security_match));
-	memset(m_secure_memory, 0, sizeof(m_secure_memory));
-	memset(m_days_left, 0, sizeof(m_days_left));
-	memset(m_start_time, 0, sizeof(m_start_time));
+	std::fill_n(m_unique_pattern, sizeof(m_unique_pattern), 0);
+	std::fill_n(m_identification, sizeof(m_identification), 0);
+	std::fill_n(m_security_match, sizeof(m_security_match), 0);
+	std::fill_n(m_secure_memory, sizeof(m_secure_memory), 0);
+	std::fill_n(m_days_left, sizeof(m_days_left), 0);
+	std::fill_n(m_start_time, sizeof(m_start_time), 0);
```

Comment:
Please use `std::size` to get the number of elements, rather than `sizeof` which gets the number of bytes (e.g. `std::fill_n(foo, std::size(foo), 0);`).